libkbfs/md_journal.go

```diff
@@ -1178,7 +1226,11 @@ func (j *mdJournal) clear(
 	// Garbage-collect the old branch entries.  TODO: we'll eventually
 	// need a sweeper to clean up entries left behind if we crash
 	// here.
+	isLocalSquash := bid == LocalSquashBranchID
```

Comment:
This is removing the MD data itself, not the entries.  `j.j.clear()` does that above, correctly.  We don't want to clear the MD data for the local squashes, because they will be reused in the resolved journal as is.  I can add a comment explaining that.

As for the BID check, that seems correct to me.  The head (i.e., the latest MD in the journal) should always have the right branch ID, regardless of squashiness.